cpp/src/Manager.cpp

```diff
@@ -315,44 +301,25 @@ bool Manager::RemoveDriver
 		string const& _controllerPath
 )
 {
-	// Search the pending list
-	for( list<Driver*>::iterator pit = m_pendingDrivers.begin(); pit != m_pendingDrivers.end(); ++pit )
-	{
-		if( _controllerPath == (*pit)->GetControllerPath() )
-		{
-			delete *pit;
-			m_pendingDrivers.erase( pit );
-			Log::Write( LogLevel_Info, "mgr,     Driver for controller %s removed", _controllerPath.c_str() );
-			return true;
-		}
-	}
-
 	// Search the ready map
 	for( map<uint32,Driver*>::iterator rit = m_readyDrivers.begin(); rit != m_readyDrivers.end(); ++rit )
 	{
 		if( _controllerPath == rit->second->GetControllerPath() )
 		{
-			/* data race right here:
```

Comment:
This work doesn't fix the race condition described here. Please restore this comment.